hypha/apply/projects/views/project.py

```diff
@@ -717,23 +702,33 @@ def get(self, request, *args, **kwargs):
             raise
         return response
 
-    def get_paf_data_with_field(self, project):
-        data_dict = {}
-        form_data_dict = project.form_data
-        for field in project.form_fields.raw_data:
-            if field['id'] in form_data_dict.keys():
-                if isinstance(field['value'], dict) and 'field_label' in field['value']:
-                    data_dict[field['value']['field_label']] = form_data_dict[field['id']]
-
-        return data_dict
-
-    def get_supporting_documents(self, project):
-        documents_dict = {}
-        for packet_file in project.packet_files.all():
-            documents_dict[packet_file.title] = self.request.build_absolute_uri(
-                reverse('apply:projects:document', kwargs={'pk': project.id, 'file_pk': packet_file.id})
-            )
-        return documents_dict
+
+@method_decorator(staff_or_finance_or_contracting_required, name='dispatch')
+class ProjectDetailDocxView(SingleObjectMixin, View):
+    model = Project
+
+    def get(self, request, *args, **kwargs):
+        self.object = self.get_object()
```

Comment:
I would recommend merging the two views related to downloading the pdf and docx together. 

Move the `get_paf_download_context` in the view func itself, it's logic in it is related to the view, so it doesn't need to be extracted out. 

Something on the lines below:

```python
class ProjectExportView(SingleObjectMixing...):
  model = Project

  def get(self, request, export_type):
     project = self.get_object()
     context = ...

     if export_type = "pdf":
        return render_as_pdf(template="template.html", filename=..., context)

     if export_type = "docx":
        context["show_footer"] = False
        return render_as_docx(template="template.html", filename=..., context)
```

Reply:
Done!